modules/categoryTranslation.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * This module translates iab category to freewheel industry using translation mapping file
+ * Publisher can set translation file by using setConfig method
+ *
+ * Example:
+ * config.setConfig({
+ *    'brandCategoryTranslation': {
+ *      'translationFile': 'http://sample.com'
+ *    }
+ * });
+ * If publisher has not defined translation file than prebid will use default prebid translation file provided here <TODO add url once it is uploaded on cdn>
+ */
+
+import { config } from '../src/config';
+import { hooks, hook } from '../src/hook';
+import { ajax } from '../src/ajax';
+import { timestamp, logError, setDataInLocalStorage, getDataFromLocalStorage } from '../src/utils';
+
+// TODO udpate url once it is uploaded on cdn
+const DEFAULT_TRANSLATION_FILE_URL = 'https://api.myjson.com/bins/j5d0k';
+const DEFAULT_IAB_TO_FW_MAPPING_KEY = 'iabToFwMappingkey';
+const DEFAULT_IAB_TO_FW_MAPPING_KEY_PUB = 'iabToFwMappingkeyPub';
+const refreshInDays = 1;
+
+let adServerInUse;
+export const registerAdserver = hook('async', function(adServer) {
+  adServerInUse = adServer;
+}, 'registerAdserver');
+
+export function getAdserverCategoryHook(fn, adUnitCode, bid) {
+  if (!bid) {
+    return fn.call(this, adUnitCode); // if no bid, call original and let it display warnings
+  }
+  if (!adServerInUse) {
+    registerAdserver();
+  }
+
+  let localStorageKey = (config.getConfig('brandCategoryTranslation.translationFile')) ? DEFAULT_IAB_TO_FW_MAPPING_KEY_PUB : DEFAULT_IAB_TO_FW_MAPPING_KEY;
+
+  if (bid.meta && !bid.meta.adServerCatId) {
+    let mapping = getDataFromLocalStorage(localStorageKey);
+    if (mapping) {
+      try {
+        mapping = JSON.parse(mapping);
+        mapping = mapping['data'];
+      } catch (error) {
+        logError('Failed to parse translation mapping file');
+      }
+      if (bid.meta) {
+        bid.meta.adServerCatId = (bid.meta.iabSubCatId && mapping[adServerInUse] && mapping[adServerInUse]['mapping']) ? mapping[adServerInUse]['mapping'][bid.meta.iabSubCatId] : undefined;
+      }
+    } else {
+      logError('Translation mapping data not found in local storage');
+    }
+  }
+  fn.call(this, adUnitCode, bid);
+}
+
+export function initTranslation(...args) {
+  hooks['addBidResponse'].before(getAdserverCategoryHook, 50);
```

Comment:
This is hook add part needs to be checked if it's already been setup.  When using the `setConfig` option, this function is being setup twice.